lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,213 @@
+            )
+
+        """
+        server_rest_url = config.get("results", "server_rest_url")
+        response = requests.get(f"{server_rest_url}/host_info")
+        if response.ok:
+            logger.error(
+                "Unable to determine results host info from %s/host_info",
+                server_rest_url,
+            )
+            sys.exit(1)
+        if response.text.startswith("MESSAGE"):
+            message = response.text.split("===")[1]
+            logger.info("*** Message from sysadmins of %s:", results_webserver)
+            logger.info("***\n*** %s", message)
+            logger.info("***\n*** No local actions taken.")
+            sys.exit(1)
+        results_path_prefix = response.text.split(":")[1]
+        if not results_path_prefix:
+            logger.error(
+                "fetch results host info did not contain a path prefix: %s",
+                response.text,
+            )
+            sys.exit(1)
+        """
         temp_dir = tempfile.mkdtemp(
             dir=config.pbench_tmp, prefix="pbench-move-results."
         )
-    except Exception:
-        logger.error("Failed to create temporary directory")
-        sys.exit(1)
-
-    runs_copied = 0
-    failures = 0
-
-    for dirent in config.pbench_run.iterdir():
-        if not dirent.is_dir():
-            continue
-        if dirent.name.startswith("tools-") or dirent.name == "tmp":
-            continue
-        result_dir = dirent
-        mrt = MakeResultTb(result_dir, temp_dir, _user, _prefix, config, logger)
-        result_tb_name = mrt.make_result_tb()
-        assert (
-            result_tb_name
-        ), "Logic bomb!  make_result_tb() always returns a tar ball name"
-        crt = CopyResultTb(controller, result_tb_name, config, logger)
-        crt.copy_result_tb()
-        try:
-            # We always remove the constructed tar ball, regardless of success
-            # or failure, since we keep the result directory below on failure.
-            os.remove(result_tb_name)
-            os.remove(f"{result_tb_name}.md5")
-        except OSError:
-            logger.error("rm failed to remove %s and its .md5 file", result_tb_name)
-            sys.exit(1)
 
-        try:
-            shutil.rmtree(result_dir)
-        except OSError:
-            logger.error("rm failed to remove the %s directory hierarchy", result_dir)
-            sys.exit(1)
+        runs_copied = 0
+        failures = 0
+        no_of_tb = 0
+
+        for dirent in config.pbench_run.iterdir():
+            if not dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+
+            no_of_tb += 1
+            result_dir = dirent
+            mrt = MakeResultTb(result_dir, temp_dir, config, logger)
+
+            try:
+                mrt.check_result_target_dir(result_dir, "Result")
+                mrt.check_result_target_dir(temp_dir, "Target")
```

Comment:
Why expose these methods?  Why not have that done as part of the constructor?

Reply:
I will add in constructor :)